File: ios-app/ios-app/Core/Storage/CacheManagerImpl.swift
//
//  CacheManagerImpl.swift
//  ios-app
//
//  Created by Koji Osugi on 15/01/23.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import shared

class CacheManagerImpl: CacheManager {
    
    func getString(key: String, defaultValue: String = "") -> String {
        let defaults = UserDefaults.standard
        return defaults.string(forKey: key) ?? defaultValue
    }
    
    func putString(key: String, value: String) {
        let defaults = UserDefaults.standard
        defaults.set(value, forKey: key)
    }
    
    func getBoolean(key: String, defaultValue: Bool) -> Bool {
        let defaults = UserDefaults.standard
        return defaults.bool(forKey: key)
    }
    
    func putBoolean(key: String, value: Bool) {
        let defaults = UserDefaults.standard
        defaults.set(value, forKey: key)
    }
}

File: android-app/src/main/java/com/kurrency/android/core/CacheManagerImpl.kt
package com.kurrency.android.core

import com.kurrency.data.local.storage.CacheManager

class CacheManagerImpl: CacheManager {

    override fun putString(key: String, value: String) {

    }

    override fun getString(key: String, defaultValue: String): String {
        return ""
    }

    override fun putBoolean(key: String, value: Boolean) {

    }

    override fun getBoolean(key: String, defaultValue: Boolean): Boolean {
        return false
    }
}